gtpv2/message/change-notification-req.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2019-2022 go-gtp authors. All rights reserved.
+// Use of this source code is governed by a MIT-style license that can be
+// found in the LICENSE file.
+
+package message
+
+import "github.com/wmnsk/go-gtp/gtpv2/ie"
+
+// ChangeNotificationRequest is a ChangeNotificationRequest Header and its IEs above.
+type ChangeNotificationRequest struct {
+	*Header
+	IMSI                             *ie.IE
+	MEI                              *ie.IE
+	IndicationFlags                  *ie.IE
+	RATType                          *ie.IE
+	ULI                              *ie.IE
+	UCI                              *ie.IE
+	PGWS5S8IPAddressForControlPlane  *ie.IE
+	LinkedEBI                        *ie.IE
+	PresenceReportingAreaInformation []*ie.IE
+	MOExceptionDataCounter           *ie.IE
+	SecondaryRATUsageDataReport      []*ie.IE
+	PrivateExtension                 *ie.IE
+	AdditionalIEs                    []*ie.IE
+}
+
+// NewChangeNotificationRequest creates a new ChangeNotificationRequest.
+func NewChangeNotificationRequest(teid, seq uint32, ies ...*ie.IE) *ChangeNotificationRequest {
+	c := &ChangeNotificationRequest{
+		Header: NewHeader(
+			NewHeaderFlags(2, 0, 1),
+			MsgTypeChangeNotificationRequest, teid, seq, nil,
+		),
+	}
+
+	for _, i := range ies {
+		if i == nil {
+			continue
+		}
+		switch i.Type {
+		case ie.IMSI:
+			c.IMSI = i
+		case ie.MobileEquipmentIdentity:
+			c.MEI = i
+		case ie.Indication:
+			c.IndicationFlags = i
+		case ie.RATType:
+			c.RATType = i
+		case ie.UserLocationInformation:
+			c.ULI = i
```

Comment:
Double check if ULI is handled correctly. In Create Req message it has `ULIForSGW` and differentiated by instance type.